### PR TITLE
Added tabs for configuration files in the documentation

### LINKED ISSUE #8384: Show configuration toml files in tabs within the docs
To better understand what I mean see [here](https://hatch.pypa.io/latest/config/environment/overview/#environment-configuration) and below in the hatch docs which also uses Material for Mkdocs.

image

It doesn't seem the case in the hatch docs that if you select one such tab that all the tabs that follow automatically select the same tab. I don't know how it is done but that is the case in the hugo docs for example [here](https://gohugo.io/getting-started/configuration/#configuration-directory) and below and it is a really nice feature.

image

I think this would be a great improvement to the `ruff` docs in terms of readability and understanding for users.

### PR DESCRIPTION
**Pull request by trag1c**
## Summary

Closes #8384.

## Test Plan

Checked whether it renders properly on the `mkdocs serve` preview.

### CODE REVIEW

**Comment by trag1c** [comment]
It seems like syntax required by the tabs plugin goes against `mdformat`'s requirements. Should the hook be removed from pre-commit?

File: crates/ruff_dev/src/generate_options.rs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 //! Used for <https://docs.astral.sh/ruff/settings/>.
 use std::fmt::Write;
 
+use ruff_python_trivia::textwrap;
```
Comment (charliermarsh) [inline crates/ruff_dev/src/generate_options.rs:6]: Ohh I didn't realize when you asked this that it was specifically for Ruff :joy:

Comment (trag1c) [reply crates/ruff_dev/src/generate_options.rs:6]: :D

**Review by charliermarsh — APPROVED** [review]
This is excellent, thank you so much!

**Comment by charliermarsh** [comment]
(Fixing Clippy, which I broke.)

**Comment by doolio** [comment]
Great job @trag1c. This makes it so much better for me as a beginner.

**Pull request merged**